### PR TITLE
Data Wrangling: Support an option to summarize the count by Token, not by Document ID

### PR DESCRIPTION
**Pull request by hidekoji**
# Description

As for do_tokenize_icu, change it to show summarized count by token if drop argument is TRUE and keep_cols argument is FALSE.

# Checklist
Make sure you have performed following items before submitting this pull request.
If not, please describe the reason.  

- [x] Add test cases for this fix/enhancement
- [ ] Pass devtools::check()
- [ ] Pass devtools::test()
- [ ] Test installing from github
- [x] Tested with Exploratory

### CODE REVIEW

File: R/string_operation.R [hunk 1]
```diff
@@ -241,7 +244,16 @@ do_tokenize_icu <- function(df, text_col, token = "word", keep_cols = FALSE,
   }
   # result column order should be document_id, <token_col>, <count_col> ...
   if(!with_id) {
-    result %>% select(!!rlang::sym(token_col), !!rlang::sym(count_col), dplyr::everything())
+    result <-result %>% select(!!rlang::sym(token_col), !!rlang::sym(count_col), dplyr::everything())
+    if(drop && !keep_cols && summary_level == "token") { # if the summary_level is token and resulting columns are only token and count, summarize it by token.
+      result <- result %>% dplyr::group_by(!!rlang::sym(token_col)) %>% dplyr::summarise(!!rlang::sym(count_col) := n())
```
Comment (hideaki) [inline R/string_operation.R:249]: This should be sum rather than n().

File: R/string_operation.R [hunk 2]
```diff
@@ -241,7 +244,16 @@ do_tokenize_icu <- function(df, text_col, token = "word", keep_cols = FALSE,
   }
   # result column order should be document_id, <token_col>, <count_col> ...
   if(!with_id) {
-    result %>% select(!!rlang::sym(token_col), !!rlang::sym(count_col), dplyr::everything())
+    result <-result %>% select(!!rlang::sym(token_col), !!rlang::sym(count_col), dplyr::everything())
+    if(drop && !keep_cols && summary_level == "token") { # if the summary_level is token and resulting columns are only token and count, summarize it by token.
```
Comment (hideaki) [inline R/string_operation.R:248]: Conditions other than summary_level seems unnecessary.

File: R/string_operation.R [hunk 3]
```diff
@@ -170,7 +172,8 @@ do_tokenize_icu <- function(df, text_col, token = "word", keep_cols = FALSE,
                                  remove_hyphens = FALSE, remove_separators = TRUE,
                                  remove_symbols = TRUE, remove_twitter = TRUE,
                                  remove_url = TRUE, stopwords_lang = NULL,
-                                 hiragana_word_length_to_remove = 2, ...){
+                                 hiragana_word_length_to_remove = 2,
+                                 summary_level = "document", sort_by = "count", ...){
```
Comment (hideaki) [inline R/string_operation.R:176]: summary_level:document/token should be all/row.

File: R/string_operation.R (hunk 1 above)
Comment (hidekoji) [reply R/string_operation.R]: fixed it to use sum() instead of n()

File: R/string_operation.R (hunk 2 above)
Comment (hidekoji) [reply R/string_operation.R]: Updated as suggested.

File: R/string_operation.R (hunk 3 above)
Comment (hidekoji) [reply R/string_operation.R]: updated as suggested.

**Pull request merged**